modules/alloydb/main.tf

```diff
@@ -0,0 +1,372 @@
+      backup_window = var.automated_backup_configuration.backup_window
+      labels        = var.labels
+
+      dynamic "encryption_config" {
+        for_each = var.encryption_config != null ? { 1 = 1 } : {}
+        content {
+          kms_key_name = var.encryption_config.primary_kms_key_name
+        }
+      }
+
+      weekly_schedule {
+        days_of_week = var.automated_backup_configuration.weekly_schedule.days_of_week
+        start_times {
+          hours   = var.automated_backup_configuration.weekly_schedule.start_times.hours
+          minutes = var.automated_backup_configuration.weekly_schedule.start_times.minutes
+          seconds = var.automated_backup_configuration.weekly_schedule.start_times.seconds
+          nanos   = var.automated_backup_configuration.weekly_schedule.start_times.nanos
+        }
+      }
+
+      dynamic "quantity_based_retention" {
+        for_each = var.automated_backup_configuration.retention_count != null ? { 1 = 1 } : {}
+        content {
+          count = var.automated_backup_configuration.retention_count
+        }
+      }
+
+      dynamic "time_based_retention" {
+        for_each = var.automated_backup_configuration.retention_period != null ? { 1 = 1 } : {}
+        content {
+          retention_period = var.automated_backup_configuration.retention_period
+        }
+      }
+    }
+  }
+
+  dynamic "continuous_backup_config" {
+    for_each = var.continuous_backup_configuration.enabled ? { 1 = 1 } : {}
+    content {
+      enabled              = true
+      recovery_window_days = var.continuous_backup_configuration.recovery_window_days
+      dynamic "encryption_config" {
+        for_each = var.encryption_config != null ? { 1 = 1 } : {}
+        content {
+          kms_key_name = var.encryption_config.primary_kms_key_name
+        }
+      }
+    }
+  }
+
+  dynamic "encryption_config" {
+    for_each = var.encryption_config != null ? { 1 = 1 } : {}
+    content {
+      kms_key_name = var.encryption_config.primary_kms_key_name
+    }
+  }
+
+  dynamic "initial_user" {
+    for_each = var.initial_user != null ? { 1 = 1 } : {}
+    content {
+      user     = var.initial_user.user
+      password = var.initial_user.password
+    }
+  }
+
+  dynamic "maintenance_update_policy" {
+    for_each = var.maintenance_config.enabled ? { 1 = 1 } : {}
+    content {
+      maintenance_windows {
+        day = var.maintenance_config.day
+        start_time {
+          hours   = var.maintenance_config.start_times.hours
+          minutes = var.maintenance_config.start_times.minutes
+          seconds = var.maintenance_config.start_times.seconds
+          nanos   = var.maintenance_config.start_times.nanos
+        }
+      }
+    }
+  }
+}
+
+resource "google_alloydb_instance" "primary" {
+  cluster           = google_alloydb_cluster.primary.id
+  availability_type = var.availability_type
+  database_flags    = var.flags
+  display_name      = "${local.prefix}${var.name}"
```

Comment:
I'd us this as a default and provide a field for the user to override

Reply:
Updated!